platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,381 @@
+  - We have a safeguard that mostly works; if we run into a failure using the
+    latest CMS content, we try again with the cached content in S3 from the last
+    successful build
+    - This doesn't work when the GraphQL query has been updated; that
+      invalidates the cache
+    - Errors in liquid templates would potentially break the build and falling
+      back to the cache won't help
+
+### High Level Design
+The current front end build will be split up into two distinct builds:
+1. Content build
+    - Input: CMS content
+    - Output: HTML files, assets from the CMS (PDFs, images and the like)
+1. Application build
+    - Input: `vets-website` code
+    - Output: JavaScript and CSS bundles
+
+The output of both these builds will be deployed to separate buckets. The
+reverse proxies will route traffic intelligently to the appropriate bucket given
+the request.
+
+![Proposed traffic flow to S3 buckets](images/proposed-traffic-flow-to-s3-buckets.png)
+
+**Important note:** JavaScript application landing pages will live in the CMS.
+To get an application into production, the page in Drupal will need to be
+published.
+
+**Another important note:** There is no way to coordinate these two deploys to
+make an application live for the first time. The process will be to manually:
+1. Verify the application assets are live in production
+1. Publish the landing page
+1. Deploy a new content build
+
+## Specifics
+
+### Detailed Design
+
+#### Build and deploy
+- The content build will live in a separate repository
+  - **Question:** What should we call this?
+  - **Question:** Can the application build be run in Circle?
+  - **Question:** Where can we run the content build?
+- Content will be deployed to a separate S3 bucket from the application code
+- The S3 buckets will be [versioned](https://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html)
+  in them with the contents of each deployment
+  - Old directories will be removed after 90 days
+    - **Question:** What makes sense? This is a totally arbitrary number
+
+#### Routing
+- The nginx will route their traffic based on some configuration file
+  - **Question:** What configuration file? How does it work?
+
+#### Testing changes to applications
+- The Webpack configuration will use
+  [`html-webpack-plugin`](https://github.com/jantimon/html-webpack-plugin) to
+  generate temporary landing pages for React applications
+  - This will happen for all build types
+    - Engineers will be able to access the site on a `localhost` build
+    - CI can run E2E tests against production code without the output of the
+      content build
+- The E2E test script in CI will start a process to serve these pages so the
+  tests can run a browser and access the applications
+- **Important note:** These HTML files will not be served in production; the
+  landing pages for the applications come from the content build
+  - **Question:** How do we make this happen?
```

Comment:
I'm thinking something vaguely like putting the HTML files in a `build/` or `landing-pages/` or something, and the "real" Webpack assets (JS, CSS, etc.) in like a `dist/` or something.

But then how do we go from there to either
1. Not pushing the temporary landing pages to the S3 bucket, OR
1. Not routing to them when the request for the application page is made